New Search Homepage

youriwims opened the pull request:
# 👩‍🎨  READY FOR DESIGN REVIEW
## 🤓 Master PR for new search homepage
### 👶 Child PRs 👉 #1102, #1103, #1104

Closes #1039 
Closes #1041 
Closes #1077

Related Tickets: #1039, #1041, #1059

_This PR takes care of the nav + search/dropdown style revisions & homepage redirect._

View Updates Here 👉 https://network-pulse-staging-pr-1101.herokuapp.com/

cc/ @xmatthewx @natalieworth @alanmoo @mmmavis

alanmoo commented:
A few thoughts:

1. I thought the page was broken until I looked into the code. At the very least, pressing "return" needs to perform a search. But why are we changing the existing behavior?

2. This doesn't close #1041 as it stands; that ticket asks for the search bar on the homepage.

3. What is this about? `Testing Branch (ref): #1096`

alanmoo commented on pages/search/search.scss:
```diff
@@ -16,4 +16,22 @@
     align-items: center;
   }
+
+  .search-bar input {
```
Is this supposed to _only_ target the search bar, or should it be an update for all form fields?

youriwims replied:
Just checked the designs, this should be for all form fields.I'll update this.

youriwims commented:
> * I thought the page was broken until I looked into the code. At the very least, pressing "return" needs to perform a search. But why are we changing the existing behavior?

Just talked with @natalieworth about this and she agrees that using "return" should be included as well. But she still has to follow up w/Matthew on final search functionality. I'm just going to add the "return" functionality here so that he can review when he's back.
> * This doesn't close #1041 as it stands; that ticket asks for the search bar on the homepage.

True...updated initial comment.

> * What is this about? `Testing Branch (ref): #1096`

It was just a link to my testing branch where I originally made the updates while the dropdown was still in review before it landed on master.

natalieworth commented:
@youriwims YES this is looking awesome!

- [ ] 1. Overall design is good 
Yes, design on mobile, tablet and desktop looks great. It's basically the mockups. So good!
- [ ] 2. Search looks & functions as expected
Yep! Search doesn't feel weird with the People/ Projects filtering. All good.
- [ ] 3. Dropdown looks & functions as expected
Dropdown and form styles of both are great. I know the drop down is using the standard label styling but aside from that everything looks good.
- [ ] 4. Nav looks & functions as expected (when signed in & signed out)
Styling of the mobile, desktop, and tablet nav all looks good. I think the only thing I noticed for the function is for the 'add new' button (it still looks like it is using the old purple on click):
<img width="1214" alt="Screen Shot 2019-07-05 at 1 10 57 PM" src="https://user-images.githubusercontent.com/14876567/60743451-89bdc600-9f26-11e9-8603-741b60d8a247.png">

natalieworth commented:
@youriwims Oh, another functionality thing that I ran into: 
I'm not able to click on the search/ help selection when I test it on smaller screen sizes in the firefox developer tools window.

natalieworth commented:
@youriwims I think the spacing is good too now. It looked tight before when the search/ help wasn't there—but it's looking pretty good/ close to the comps, so no need for new tic.

natalieworth commented:
- [ ] 1. @youriwims Sorry, few more things/ questions after I had design team test it too:
- [ ] 2. [These pages (when you click on the help tag within the entry page)](https://network-pulse-staging-pr-1101.herokuapp.com/help/stand-together) could link to [this page](https://network-pulse-staging-pr-1101.herokuapp.com/projects?helpType=Stand%20together) Kristina mentioned that it's similar to what Alan told her to do [in this ticket](https://github.com/mozilla/foundation.mozilla.org/issues/3165#issuecomment-508547752)
- [ ] 3. For the help dropdown type, is there one lighter weight we could go down to? I think I had bold in my mockups but it's feeling a bit heavy on the screen. 
- [ ] 4. Make 'n' lowercase in the 'Add new' button

***Also just noting that there is this page (I thought we were going to get rid of it, but need to double check and Matthew's out for pto):
<img width="1678" alt="Screen Shot 2019-07-05 at 5 15 22 PM" src="https://user-images.githubusercontent.com/14876567/60748997-93582580-9f48-11e9-9774-85c3da9a8760.png">

youriwims commented:
@natalieworth  ⬇️ 
> Styling of the mobile, desktop, and tablet nav all looks good. I think the only thing I noticed for the function is for the 'add new' button (it still looks like it is using the old purple on click):

Just fixed this. I wasn't sure if you wanted an 'active button'. But prod has one and just to test it out and see if you like it, it's just black background and white text. It's an easy change if you'd rather it not have an active state when clicked.
> I'm not able to click on the search/ help selection when I test it on smaller screen sizes in the firefox developer tools window.

Sorry about that weird bug, should be all fixed now.

>For the help dropdown type, is there one lighter weight we could go down to? I think I had bold in my mockups but it's feeling a bit heavy on the screen.

I dropped the weight from 700 to 600. I don't notice much of a difference atm haha but I may have been staring at it too long. Let me know if you just want it to go to normal weight(400).

> ***Also just noting that there is this page (I thought we were going to get rid of it, but need to double check and Matthew's out for pto):

Yeahhhh, ummm in the meantime, maybe Alan can help out with this. @alanmoo what page should the entry tags link to for now or should we remove the link?

Btw @natalieworth everything else that I didn't mention here, I just checked off as I completed it 👍

natalieworth commented:
Sweet, looking good @youriwims I think the only thing is I'd maybe remove the black and white text click state that stays on the 'add new' is click, just so it's up to date with brand standards. 

I'm going to mockup the issue areas and also I noticed the tags too could have the intro stuff at the top since now we're only keeping the homepage. Should I open a new ticket for those based on what Alan said this morning @youriwims ?

Mobile/ tablet function + design looks nice too!

youriwims commented:
Sure @natalieworth , creating a new ticket with the revised issue mocks would be helpful. If it's a few things, can you point out the differences updates? As for the issues & tags, when a user clicks them from an entry, do you still want them to be redirected to the homepage as a 'searched' item?

natalieworth commented:
@youriwims I think on click, the button still is remaining black.

youriwims commented:
How about now @natalieworth?

youriwims commented:
Updated temp issue design (https://redpen.io/zka3e0ac2ac566af59). @natalieworth I didn't see the mock on abstract, so I'm not exactly sure if all the specs are exactly correct since I eyeballed it 👁. Also, when you file follow-up tickets for tag-related bits & the issues page, can you reference this pr, so it's all connected? 😊

Pomax commented on scss/foundation-styles/buttons.scss:
```diff
@@ -83,8 +83,8 @@
-// append external link glyph
-.btn-secondary[href*="//"]:not([href*="foundation.mozilla.org"]) {
+// // append external link glyph
+.btn-secondary[href*="//"]:not([href*="test.example.com"]):not([href*="pulse-api.mofostaging.net"]):not([href*="api.mozillapulse.org"]) {
```
this feels like we're using the wrong approach to determine whether something is an external link or not.

youriwims replied:
I think this may have been a temporary solution. I'm open to suggestions 😅 cc/ @mmmavis

alanmoo commented on server.js:
```diff
@@ -38,7 +38,7 @@ function renderPage(appHtml, reactHelmet) {
         <link rel="manifest" href="/manifest.json">
         <link rel="stylesheet" type="text/css" href="https://code.cdn.mozilla.net/fonts/zilla-slab.css">
-        <link rel="stylesheet" href="https://fonts.googleapis.com/css?family=Nunito+Sans:400,300,700,300i">
+        <link rel="stylesheet" href="https://fonts.googleapis.com/css?family=Nunito+Sans:300,300i,400,600,700">
```
Do we need the 300 weights? A command+F doesn't find that being used

youriwims replied:
Really?...I'm using it over in #1059 & it's used in the little capsule when you search a help cat. on the homepage. As for 300i, I guess that can be removed.

alanmoo replied:
Ah, that's probably why it didn't show up in a search of _that page_. Obviously they're fine to keep if they're used 👍

alanmoo commented on scss/mixin.scss:
```diff
@@ -43,3 +43,20 @@
   }
 }
+
+@mixin form-control( $border-color ) {
```
I'm confused by this- `.form-control` is a class bootstrap gives us, so making a mixin with the same name that may or may not override things is...hard to understand. If we don't want this styling applied to everything, would this better be suited to an additional class on certain elements?

youriwims replied:
the main difference here is the border color. would it be easiest if these styles were just moved to `.form-control` and border-color was overrided specifically on the search and dropdown elements?

alanmoo replied:
Overriding bootstrap's `.form-control` at a project level is a better solution here if we want all these styles to apply to all form elements. If we just want them to apply to these elements, at least pick a different name?

youriwims replied:
👍

alanmoo commented on scss/mixin.scss:
```diff
@@ -43,3 +43,20 @@
+    border-color: $light-blue;
+  }
+
+  @media (min-width: $bp-md) {
```
Following the above thought, a media query on a generic mixin feels ripe for bugs

youriwims replied:
See https://github.com/mozilla/network-pulse/pull/1101#discussion_r302239629

alanmoo commented on pages/search/search.scss:
```diff
@@ -16,4 +16,40 @@
   }
+
+  .help-dropdown {
+    @include form-control( $black );
```
Not to beat a dead horse, but this is the styling that actually gets built here- it looks like this mixin is effectively only applying a border-radius and border-color at larger sizes. Something just doesn't feel right.
<img width="293" alt="image" src="https://user-images.githubusercontent.com/1682681/60998292-9d9e6900-a326-11e9-8ac5-8a16dd46f614.png">

alanmoo replied:
Oh, I see where the extra styles come in. It's a little less weird now that I realize that.

youriwims replied:
yeah, you're right, the mixin doesn't play a huge role here and can be removed.

alanmoo commented on components/search-result-counter.jsx:
```diff
@@ -21,7 +22,7 @@ export default (props) => {
-
-  return <p className="query-text">{counterText}</p>;
+  }
+  
```
Our linter doesn't yell about this extra space?

```suggestion

```

youriwims commented:
@alanmoo the form-control thing is fixed, but I'm still getting that console error. cc: @Pomax

natalieworth commented:
Looks good @youriwims
